store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -97,13 +106,25 @@ private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> oper
 
     }
 
-    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation operation, final FederatedStore store) throws OperationException {
+    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation<INPUT, OUTPUT> operation, final FederatedStore store) throws OperationException {
         try {
             Object rtn = null;
 
             final BiFunction mergeFunction = getMergeFunction(operation, store, context, isEmpty(resultsFromAllGraphs));
+            final List<GraphSerialisable> graphs = getGraphs(operation, context, store);
+
+            // If default merging and only have one graph or no common groups then just return the current results
+            if (!graphs.isEmpty()
+                    && mergeFunction instanceof ApplyViewToElementsFunction
+                    && (graphs.size() == 1 || !graphsHaveCommonSchemaGroups(graphs))) {
+                LOGGER.info("Skipping merge function as not required when only one graph or no common groups");
+                // Just flatten current results that are essentially a list of lists into one list and return
+                HashSet<Element> mergedResults = new HashSet<>();
+                resultsFromAllGraphs.forEach(result -> ((Iterable<Element>) result).iterator().forEachRemaining(mergedResults::add));
```

Comment:
We should instead use the concat merge function which exists for this. This is also inefficient as it iterates through everything which does not preserve lazy behaviour (see #2819). Would be easiest to swap out the `mergeFunction` and use the existing code below.

Reply:
This is something I noticed after opening, I believe moving to `flatMap()` rather than concat is preferable for this use case as there might be more than two graphs. Updated to now return a lazy iterable from the merged streams.

Reply:
That is better, but functionally the same as using the concat iterator for what I can tell.

Reply:
I don't know of any reason. The concat merge function is simple as the merging is done with other classes which are also used elsewhere in Gaffer. Potentially this could all be refactored to use streams, but I don't see many benefits. `new ChainedIterable<>(item1, item2)` is easier to read than a multiline stream statement.

Reply:
Streams do have a lot of advantages though as spliterators have a much lower per element cost than iterators and reduced memory bus activity to implement for the merge function its as easy as something like:

```java
Stream<Object> stateStream = StreamSupport.stream(state.spliterator(), false);
Stream<Object> updateStream = StreamSupport.stream(updateSafe.spliterator(), false);
Iterable<Object> merged = () -> Stream.concat(stateStream, updateStream).iterator();
return merged;
```

Reply:
Could be worth raising a tech debt ticket to refactor all of this (total rework/remove of these different classes) with some notes about streams. For this issue/PR we should use the preexisting merge functions and not introduce a mix of two different approaches.